tokio/src/stream/stream_map.rs

```diff
@@ -0,0 +1,435 @@
+use crate::stream::Stream;
+
+use std::borrow::Borrow;
+use std::hash::Hash;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// Combine many streams into one, indexing each source stream with a unique
+/// key.
+///
+/// `StreamMap` is similar to [`StreamExt::merge`] in that it combines source
+/// streams into a single merged stream that yields values in the order that
+/// they arrive from the source streams. However, `StreamMap` has a lot more
```

Comment:
"in the order that they arrive" - if I understand the implementation correctly this may be misleading: if multiple stream become Ready close enough to the same time they can still be returned in any (random) order.

Reply:
Yes... but the question is whether or not the details matter in practice and is there a way to explain reality without adding a heap of confusion.